docs/examples/dependency_injection/dependency_skip_validation.py

```diff
@@ -7,7 +7,7 @@
 from litestar.params import Dependency
 
 
-def provide_str() -> str:
+async def provide_str() -> str:
     """Returns a string."""
     return "whoops"
```

Comment:
I think it's a problem to have async non-blocking examples like this, since these are not good examples of using async.

Reply:
You're right, it should not go without an explanation. How do you feel about leaving them like this now and updating the docs later? 

This whole chapter could use a re-write, so adding a section that goes a bit more in-depth about async dependencies and when and why, examples included, would fit well in there.

I'd like to not do this within the scope of this PR though if possible.

Reply:
sure